src/test/custom_nodes/node_choose_maximum.cpp

```diff
@@ -0,0 +1,163 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "../../custom_node_interface.h"
+
+static const size_t dummyInputSize = 10;
+
+extern "C" {
+enum METHOD {
+    MAXIMUM_MINIMUM,
+    MAXIMUM_AVERAGE,
+    MAXIMUM_MAXIMUM,
+    METHOD_COUNT
+};
+
+static const std::string INPUT_TENSOR_NAME = "input_tensors";
+
+int execute(const struct CustomNodeTensor* inputs, int inputsLength, struct CustomNodeTensor** outputs, int* outputsLength, const struct CustomNodeParam* params, int paramsLength) {
+    // choose selection criteria
+    uint16_t selection_method = METHOD_COUNT;
+    if (paramsLength != 1) {
+        std::cout << "Wrong number of parameters - expected 1" << std::endl;
+    }
+    for (int i = 0; i < paramsLength; i++) {
+        if (strcmp(params[i].key, "selection_criterium") == 0) {
+            if (strcmp(params[i].value, "MAXIMUM_MINIMUM") == 0) {
+                selection_method = MAXIMUM_MINIMUM;
+            } else if (strcmp(params[i].value, "MAXIMUM_MAXIMUM") == 0) {
+                selection_method = MAXIMUM_MAXIMUM;
+            } else if (strcmp(params[i].value, "MAXIMUM_AVERAGE") == 0) {
+                selection_method = MAXIMUM_AVERAGE;
+            } else {
+                std::cout << "Not allowed method chosen:" << params[i].value << std::endl;
+                return 1;
+            }
+        } else {
+            std::cout << "Non recognized param string" << std::endl;
+            return 1;
+        }
+    }
+    // TODO validate inputs
+    *outputsLength = 1;
+    *outputs = new CustomNodeTensor[*outputsLength];
+    std::cout << "Creating outputs:" << *outputs << std::endl;
+    // TODO check result
+    CustomNodeTensor& resultTensor = *outputs[*outputsLength - 1];
+    resultTensor.name = "maximum_tensor";
+    float* result = new float[dummyInputSize];
+    resultTensor.data = reinterpret_cast<uint8_t*>(result);
+    resultTensor.dimsLength = 2;
+    resultTensor.dims = new uint64_t[resultTensor.dimsLength];
+    resultTensor.dims[0] = 1;
+    resultTensor.dims[1] = dummyInputSize;
+    resultTensor.dataLength = 1 * dummyInputSize * sizeof(float);
```

Comment:
What is magic 1? I suppose this is batch size. Maybe use already defined resultTensor.dims above?